numba/cpython/randomimpl.py

```diff
@@ -191,69 +199,146 @@ def get_shifted_int(nbits):
-    res = get_next_double(context, builder, state_ptr)
-    return impl_ret_untracked(context, builder, sig.return_type, res)
-
-@glue_lowering("np.random.random")
-@glue_lowering("np.random.random_sample")
-@glue_lowering("np.random.sample")
-@glue_lowering("np.random.ranf")
-def random_impl(context, builder, sig, args):
-    state_ptr = get_state_ptr(context, builder, "np")
-    res = get_next_double(context, builder, state_ptr)
-    return impl_ret_untracked(context, builder, sig.return_type, res)
-
-
-@glue_lowering("random.gauss", types.Float, types.Float)
-@glue_lowering("random.normalvariate", types.Float, types.Float)
-def gauss_impl(context, builder, sig, args):
-    res = _gauss_impl(context, builder, sig, args, "py")
-    return impl_ret_untracked(context, builder, sig.return_type, res)
-
-
-@glue_lowering("np.random.standard_normal")
-@glue_lowering("np.random.normal")
-@glue_lowering("np.random.normal", types.Float)
-@glue_lowering("np.random.normal", types.Float, types.Float)
-def np_gauss_impl(context, builder, sig, args):
-    sig, args = _fill_defaults(context, builder, sig, args, (0.0, 1.0))
-    res = _gauss_impl(context, builder, sig, args, "np")
-    return impl_ret_untracked(context, builder, sig.return_type, res)
+@overload(random.seed)
+def seed_impl(seed):
+    if isinstance(seed, types.Integer):
+        return _seed_impl('py')
+
+
+@overload(np.random.seed)
+def seed_impl(seed):
+    if isinstance(seed, types.Integer):
+        return _seed_impl('np')
+
+
+def _seed_impl(state_type):
+    @intrinsic
+    def _impl(typingcontext, seed):
+        def codegen(context, builder, sig, args):
+            seed_value, = args
+            fnty = ir.FunctionType(ir.VoidType(), (rnd_state_ptr_t, int32_t))
+            fn = cgutils.get_or_insert_function(builder.function.module, fnty,
+                                                'numba_rnd_init')
+            builder.call(fn, (get_state_ptr(context, builder, state_type),
+                              seed_value))
+            return context.get_constant(types.none, None)
+        return signature(types.void, types.uint32), codegen
+    return lambda seed: _impl(seed)
+
+
+@overload(random.random)
+def random_impl():
+    @intrinsic
+    def _impl(typingcontext):
+        def codegen(context, builder, sig, args):
+            state_ptr = get_state_ptr(context, builder, "py")
+            return get_next_double(context, builder, state_ptr)
+        return signature(types.double), codegen
+    return lambda: _impl()
+
+
+@overload(np.random.random)
+@overload(np.random.random_sample)
+@overload(np.random.sample)
+@overload(np.random.ranf)
+def random_impl0():
+    @intrinsic
+    def _impl(typingcontext):
+        def codegen(context, builder, sig, args):
+            state_ptr = get_state_ptr(context, builder, "np")
+            return get_next_double(context, builder, state_ptr)
+        return signature(types.double), codegen
```

Comment:
Whilst aliased, these `types.double` should probably be `types.float64` to reflect the NumPy semantics of a `float64` return type.